src/org/labkey/targetedms/TargetedMSManager.java

```diff
@@ -861,6 +863,24 @@ public static TargetedMSRun getRunByLsid(String lsid, Container c)
         throw new IllegalArgumentException("More than one TargetedMS runs found for LSID "+lsid);
     }
 
+    public static boolean updateSkydDataId(ITargetedMSRun run, ExpData newSkydData, User user)
+    {
+        if (!(run instanceof TargetedMSRun) || newSkydData == null)
+        {
+            return false;
```

Comment:
Consider making this more aggressive and throw an `IllegalArgumentException` as they seem like a bigger missed assumption than the other conditions below.